app/controllers/subscription.js

```diff
@@ -201,6 +202,39 @@ export default Ember.Controller.extend({
         this.updateCartAvailability(0, cartItem);
       }
     }
+
+    // mark message read here
+    if (type === "message") {
+      var router = this.get("target");
+      var currentUrl = window.location.href.split("#").get("lastObject");
+
+      var messageRoute = this.get("messagesUtil").getRoute(data.item[type]);
```

Comment:
no real problem, but this method is really huge and complicated 😱 , if there's room for some refactoring into single-purpose methods that'd be a good improvement.